addons/test_mail/tests/test_mail_composer.py

```diff
@@ -423,6 +434,32 @@ def test_mail_composer_rights_portal(self):
             self.assertEqual(self.test_record.message_ids[0].body, '<p>Body text 2</p>')
             self.assertEqual(self.test_record.message_ids[0].author_id, portal_user.partner_id)
 
+    @users('user_rendering_restricted')
+    def test_mail_composer_rights_template(self):
+        template_1 = self.template.copy({
+            'report_name': 'TestReport for {{ object.name }}.html',  # test cursor forces html
+            'report_template': self.test_report.id,
+        })
+        attachment_data = self._generate_attachments_data(2)
+        template_1.write({
+            'attachment_ids': [(0, 0, dict(a, res_model="mail.template", res_id=template_1.id)) for a in attachment_data]
+        })
+
+        template_1_attachments = template_1.attachment_ids
+        self.assertEqual(len(template_1_attachments), 2)
+
+        composer = self.env['mail.compose.message'].with_context(
```

Comment:
Why not using the Form, just to test a "real life scenario" ?

Reply:
How do you click on the "send" button from a Form ?

Reply:
You don't. You save the form, then call the method on the underlying recordset.